### PR TITLE
1.6 PLAT 6859 support configurable latency

### PR DESCRIPTION
**Pull request by lilachmaliniak**
dynamic downloader polling interval

### CODE REVIEW

File: lib/entry/FlavorDownloader.js [hunk 1]
```diff
@@ -448,6 +450,18 @@ var FlavorDownloader = (function() {
         };
     };
 
+    function initPollingIntervals (entryObj) {
+        if (entryObj.segmentDuration) {
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: change to _.isNumber(entryObj.segmentDuration) so we won't get exceptions

File: lib/entry/LiveEntry.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ var RecordingManager = require('./../recording/RecordingManager');
 var kalturaTypes = require('./../kaltura-client-lib/KalturaTypes');
 var Diagnostics = require('../Diagnostics/Diagnostics');
 var DiagnosticsAlertsAnalyzer = require('./../Diagnostics/DiagnosticsAlertsAnalyzer');
+const flavorMinPollingInterval = config.get("flavorMinPollingInterval");
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: you don't use this here ...

File: lib/entry/FlavorDownloader.js [hunk 3]
```diff
@@ -448,6 +450,18 @@ var FlavorDownloader = (function() {
...
+            this.successPollingInterval = Math.floor(entryObj.segmentDuration/5);
+            this.failPollingInterval = Math.floor(entryObj.segmentDuration/2);
+        } else {
+            this.successPollingInterval = config.get("flavorSuccessPollingInterval");
+            this.failPollingInterval = config.get("flavorFailPollingInterval");
+        }
+
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: please add check for min,max polling

File: lib/entry/FlavorDownloader.js (hunk 1 above)
Comment (lilachmaliniak) [inline lib/entry/FlavorDownloader.js]: done

File: lib/entry/LiveEntry.js (hunk 2 above)
Comment (lilachmaliniak) [inline lib/entry/LiveEntry.js]: moved as a const to FlavorDownloader

File: lib/entry/FlavorDownloader.js (hunk 3 above)
Comment (lilachmaliniak) [inline lib/entry/FlavorDownloader.js]: added

File: lib/entry/FlavorDownloader.js [hunk 4]
```diff
@@ -70,14 +72,16 @@ var FlavorDownloader = (function() {
         this.shouldRun = true;
         this.runStatus = 'uninitialized';
         this.lastFilePathHash = null;
-        this.successPollingInterval = config.get('flavorSuccessPollingInterval');
-        this.failPollingInterval = config.get('flavorFailPollingInterval');
+        this.successPollingInterval;
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: revert

File: lib/entry/FlavorDownloader.js [hunk 5]
```diff
@@ -448,6 +452,26 @@ var FlavorDownloader = (function() {
         };
     };
 
+    function initPollingIntervals (entryObj) {
+        if (_.isNumber(entryObj.segmentDuration)) {
+            this.successPollingInterval = Math.floor(entryObj.segmentDuration/5);
+            this.failPollingInterval = Math.floor(entryObj.segmentDuration/2);
+        } else {
+            this.successPollingInterval = config.get("flavorSuccessPollingInterval");
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: remove, since it's defined above

File: lib/entry/FlavorDownloader.js [hunk 6]
```diff
@@ -448,6 +452,26 @@ var FlavorDownloader = (function() {
...
+            this.failPollingInterval = config.get("flavorFailPollingInterval");
+        }
+
+        // limit highest stream download intensity
+        this.successPollingInterval = flavorMinPollingInterval < this.successPollingInterval ? this.successPollingInterval : flavorMinPollingInterval;
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: use Math.min and Math.max

File: lib/entry/LiveEntry.js [hunk 7]
```diff
@@ -37,6 +37,7 @@ function LiveEntry(entryObject) {
     this.playlistGenerator.on('diagnosticsAlert', this.diagnostics.pushAlert.bind(this.diagnostics));
     this.recordingEnable =  entryObject.recordStatus != kalturaTypes.KalturaRecordStatus.DISABLED  &&
         config.get('recording').enable;
+
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: remove line so the file will not appear in PR

File: lib/entry/FlavorDownloader.js (hunk 4 above)
Comment (lilachmaliniak) [inline lib/entry/FlavorDownloader.js]: done

File: lib/entry/FlavorDownloader.js (hunk 5 above)
Comment (lilachmaliniak) [inline lib/entry/FlavorDownloader.js]: done

**Pull request merged**